Chore/optimize balances

w84april opened the pull request:
## Description

- Move balances logic out of useYearn
- Implemented React's useTransition hook to prevent UI blocking during balance fetches
    - Accumulates all balance updates and applies them in a single batch
    - Returns empty balances during pending state to prevent UI from freezing (idk that's probably my laptop issue but m1 max should be good enough to handle react app)

murderteeth reviewed (changes requested):
- clean up console.logs
- is there a worker? the comments mention one but i dont see it. if there's no worker, please clear up the documentation too
- are you sure useTransition makes a difference here? useTransition is designed for cases where the ui update itself is large. for example updating a table with a lot of rows.
- do you feel we could refactor this into less code? maybe using tanstack for example. my intuition is that we could. but what do you think? doesn't have to be this pr

murderteeth reviewed (changes requested):
![image](https://github.com/user-attachments/assets/302fda2a-aeaa-478d-9d52-02bf2b303e0f)

should my balance equal the sum of my holdings? maybe a bug

w84april commented:
thanks @murderteeth 
- cleaned up all worker mentions for now (might actually add one to test later)
- yep, useTransition is removed
- don't think tanstack query might be useful here, but I did refactor main `useAsyncTrigger`, `onUpdate`, `onUpdateSome` functions that were doing almost same stuff. Now they all just call the same func with slightly different params. Also actually made `onUpdate`, `onUpdateSome` work as expected (force refetch balances and update balances state). Removed one more `useContractCall` that was triggered after withdraw/deposit and was doing their job  
- as for https://github.com/yearn/yearn.fi/pull/690#pullrequestreview-2956080214 - the issue is that [this prisma vault](https://etherscan.io/address/0x11AaE8beE9b1Da827C641540D20e4e664677e06F) is calculated in total assets but not displayed in list bc we don't have `Prisma` category. So not sure what to do here, mb safe to ignore? We are calculating total deposited value out of active, migratable and retired vaults, so changing it's status won't help

<img width="590" alt="image" src="https://github.com/user-attachments/assets/c85fafcf-4064-42ad-b736-5cb13214650c" />

murderteeth reviewed (approved):
lgtm

> the issue is that [this prisma vault](https://etherscan.io/address/0x11AaE8beE9b1Da827C641540D20e4e664677e06F) is calculated in total assets but not displayed in list bc we don't have Prisma category. So not sure what to do here, mb safe to ignore?

confirmed, i tested w a different wallet and balances matched. yeah maybe we need a retired category that only shows if you have a retired balance. ask in the group. but agree, this issue isn't related to the pr
